celltype_annotation/analysis/cell-assign-sarcoma.Rmd

```diff
@@ -0,0 +1,481 @@
+anndata_file <- file.path(anndata_dir, anndata_file)
+fs::dir_create(anndata_dir)
+
+# reference files for cell marker
+ref_dir <- file.path("..", "references")
+excel_celltype_file <- file.path(ref_dir, "Cell_marker_Human.xlsx")
+panglao_file <- file.path(ref_dir, "PanglaoDB_markers_27_Mar_2020.tsv")
+
+# output matrix files 
+cell_marker_mtx_file <- file.path(ref_dir, "cell_marker_muscle_mtx.csv")
+panglao_matrix_file <- file.path(ref_dir, "panglao_connective_tissue_mtx.csv")
+
+# cellassign predictions
+cellmarker_prediction_file <- file.path(
+  anndata_dir, 
+  glue::glue("{params$library_id}_muscle.tsv")
+)
+panglao_prediction_file <- file.path(
+  anndata_dir, 
+  glue::glue("{params$library_id}_panglao_connective_tissue.tsv")
+)
+
+```
+
+
+```{r}
+# read in all annotated sce 
+processed_sce <- readr::read_rds(local_processed_sce_path)
+```
+
+```{r}
+# write out anndata 
+scpcaTools::sce_to_anndata(processed_sce, anndata_file = anndata_file)
+```
+
+```{r}
+# source helper functions
+function_file <- file.path("..", "utils", "cellassign-helper-functions.R")
+source(function_file)
+```
+
+## CellMarker2.0
+
+Below I am using the marker gene set defined in the `Human cell markers` file on [CellMarker2.0](http://117.50.127.228/CellMarker/CellMarker_download.html).
+This file contains marker genes that have been curated from published literature. 
+More information on how marker gene sets are curated can be found in the [`Data Collection` section of the original CellMarker paper](https://doi.org/10.1093/nar/gky900) and in the [`Data expansion and pre-processing` section of the CellMarker2.0 paper](https://doi.org/10.1093/nar/gkac947).
+
+Based on the table of marker genes, it looks like the main organization is by the `tissue_type` column. 
+Within each `tissue_type`, cells are separated as either from a normal or cancer sample.
+Then all cell types within each normal or cancer sample are included along with all marker genes.
+
+This means that there are duplicates of the same cell type that are found in different tissue types.
+There also will be duplicates within a tissue type if a cell type is found in both a normal and cancer sample.
+
+For example, macrophages are found in multiple tissue types so the marker gene list for macrophages may change based on the tissue type you look at.
+Additionally, macrophages are in both normal and cancer cells so a different marker gene list for macrophages in normal muscle will vary from macrophages in cancer tissue from muscle.
+All cell types that are in normal tissue are not always listed in cancer tissue and vice versa.
+
+Because we are creating a gene by cell type matrix using the ontology ID and Ensembl ID, all marker genes for a given ontology ID will be combined into a single list of marker genes for that cell type.
+However, if we filter on tissue type first and then create this matrix, then we may miss marker genes used for a given cell type that is also found in another tissue type.
```

Comment:
I wonder if this is something we can try to work around? Can we use the tissue type to get our cell type list, but then use _all_ markers for that cell type regardless of tissue? That seems potentially dicey, but also not crazy to me. I don't know how to evaluate it, except by feel.